lib/Scheduler/ready_list.cpp

```diff
@@ -120,30 +114,22 @@ ReadyList::ReadyList(DataDepGraph *dataDepGraph, SchedPriorities prirts) {
   }
 }
 
-ReadyList::~ReadyList() {
-  Reset();
-  if (prirtyLst_)
-    delete prirtyLst_;
-  if (latestSubLst_)
-    delete latestSubLst_;
-  if (keyedEntries_)
-    delete keyedEntries_;
-}
+ReadyList::~ReadyList() { Reset(); }
 
 void ReadyList::Reset() {
-  prirtyLst_->Reset();
-  latestSubLst_->Reset();
+  prirtyLst_.Reset();
+  latestSubLst_.Reset();
 }
 
-void ReadyList::CopyList(ReadyList *othrLst) {
-  assert(prirtyLst_->GetElmntCnt() == 0);
-  assert(latestSubLst_->GetElmntCnt() == 0);
-  assert(othrLst != NULL);
+void ReadyList::CopyList(ReadyList *otherList) {
+  assert(prirtyLst_.GetElmntCnt() == 0);
+  assert(latestSubLst_.GetElmntCnt() == 0);
+  assert(otherList != NULL);
 
   // Copy the ready list and create the array of keyed entries. If a dynamic
   // heuristic is not used then the second parameter should be a nullptr and the
   // array will not be created.
-  prirtyLst_->CopyList(othrLst->prirtyLst_, keyedEntries_);
+  prirtyLst_.CopyList(&otherList->prirtyLst_, keyedEntries_.data());
```

Comment:
You might also have to modify [CopyList](https://github.com/CSUS-LLVM/OptSched/blob/master/include/opt-sched/Scheduler/lnkd_lst.h#L181).

Reply:
CopyList is getting exactly the same parameters as before. I don't follow what I'd need to modify in CopyList

Reply:
Although I just realized something I need to investigate: does `keyedEntries_.data()` return `nullptr` when we need it to?

Reply:
Sorry I wasn't aware data() returns a pointer so this is good.

Since keyedEntries is now a smallVector, we could just pass the vector to CopyList and check for the size or if it isn't empty.